homeassistant/auth/providers/trusted_networks.py

```diff
@@ -163,6 +174,13 @@ def async_validate_access(self, ip_addr: IPAddress) -> None:
         ):
             raise InvalidAuthError("Not in trusted_networks")
 
+    @callback
+    def async_validate_refresh_token(
+        self, refresh_token: RefreshToken, remote_ip: Optional[str] = None
+    ) -> None:
+        """Verify a refresh token is still valid."""
+        self.async_validate_access(ip_address(remote_ip))
+
```

Comment:
This is better than before since we re-validate trusted networks on each token refresh. It could be more complete and also check trusted users here. It could have changed since the refreshtoken was created.